src/tests/ftest/control/dmg_system_reformat.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+from avocado import fail_on
+from apricot import TestWithServers
+from daos_utils import DaosCommand
+from server_utils import ServerFailed
+from command_utils import CommandFailure
+
+
+class DmgSystemReformatTest(TestWithServers):
+    # pylint: disable=too-many-ancestors
+    """Test Class Description:
+
+    Test to verify dmg storage format reformat option works as expected on a
+    DAOS system after a controlled shutdown.
+
+    :avocado: recursive
+    """
+
+    @fail_on(CommandFailure)
+    def create_pool_at_full_capacity(self):
+        """Create a pool at full capacity.
+
+        Raises:
+            CommandFailure: raise exception if pool creation fails.
+
+        """
+        host = self.hostlist_servers[0]
+
+        storage_info = self.get_dmg_command().storage_scan()
+        scm_capacity = storage_info[host]["scm"]["capacity"]
+        nvme_capacity = storage_info[host]["nvme"]["capacity"]
+
+        # Create pool object
+        self.add_pool(create=False, connect=False)
+
+        # update pool size to full capacity
+        self.pool.scm_size.update(scm_capacity)
+        self.pool.nvme_size.update(nvme_capacity)
+        self.pool.create()
+
+    @fail_on(CommandFailure)
+    @fail_on(ServerFailed)
+    def test_dmg_system_reformat(self):
+        """
+        JIRA ID: DAOS-5415
+
+        Test Description: Test dmg system reformat functionality.
+
+        :avocado: tags=all,small,pr,hw,control,sys_reformat,dmg
+        """
+        self.create_pool_at_full_capacity()
+
+        self.log.info("Disable raising an exception to check for ENOSPACE \
+                       error on pool create")
+        self.get_dmg_command().exit_status_exception = False
+
+        # Try to create second pool and check that it fails with ENOSPACE
+        self.get_pool()
+        if self.get_dmg_command().result.exit_status != 0:
+            self.log.info("Pool create failed: {}".format(
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments